examples/templates/java/README.md

```diff
@@ -0,0 +1,252 @@
+[metadata file](kafka-to-pubsub/src/main/resources/kafka_to_pubsub_metadata.json) in resources.
+
+Navigate to the template folder:
+
+```
+cd /path/to/beam/examples/templates/java/kafka-to-pubsub
+```
+
+Build the Dataflow Flex Template:
+
+```
+gcloud dataflow flex-template build ${TEMPLATE_PATH} \
+       --image-gcr-path ${TARGET_GCR_IMAGE} \
+       --sdk-language "JAVA" \
+       --flex-template-base-image ${BASE_CONTAINER_IMAGE} \
+       --metadata-file "src/main/resources/kafka_to_pubsub_metadata.json" \
+       --jar "build/libs/beam-examples-templates-java-kafka-to-pubsub-2.25.0-SNAPSHOT-all.jar" \
+       --env FLEX_TEMPLATE_JAVA_MAIN_CLASS="org.apache.beam.templates.KafkaToPubsub"
+```
+
+### Create Dataflow Job Using the Apache Kafka to Google Pub/Sub Dataflow Flex Template
+
+To deploy the pipeline, you should refer to the template file and pass the
+[parameters](https://cloud.google.com/dataflow/docs/guides/specifying-exec-params#setting-other-cloud-dataflow-pipeline-options)
+required by the pipeline.
+
+You can do this in 3 different ways:
+1. Using [Dataflow Google Cloud Console](https://console.cloud.google.com/dataflow/jobs)
+
+2. Using `gcloud` CLI tool
+    ```
+    gcloud dataflow flex-template run "kafka-to-pubsub-`date +%Y%m%d-%H%M%S`" \
+        --template-file-gcs-location "${TEMPLATE_PATH}" \
+        --parameters bootstrapServers="broker_1:9092,broker_2:9092" \
+        --parameters inputTopics="topic1,topic2" \
+        --parameters outputTopic="projects/${PROJECT}/topics/your-topic-name" \
+        --parameters outputFormat="PLAINTEXT" \
+        --parameters secretStoreUrl="http(s)://host:port/path/to/credentials" \
+        --parameters vaultToken="your-token" \
+        --region "${REGION}"
+    ```
+3. With a REST API request
+    ```
+    API_ROOT_URL="https://dataflow.googleapis.com"
+    TEMPLATES_LAUNCH_API="${API_ROOT_URL}/v1b3/projects/${PROJECT}/locations/${REGION}/flexTemplates:launch"
+    JOB_NAME="kafka-to-pubsub-`date +%Y%m%d-%H%M%S-%N`"
+
+    time curl -X POST -H "Content-Type: application/json" \
+        -H "Authorization: Bearer $(gcloud auth print-access-token)" \
+        -d '
+         {
+             "launch_parameter": {
+                 "jobName": "'$JOB_NAME'",
+                 "containerSpecGcsPath": "'$TEMPLATE_PATH'",
+                 "parameters": {
+                     "bootstrapServers": "broker_1:9091, broker_2:9092",
+                     "inputTopics": "topic1, topic2",
+                     "outputTopic": "projects/'$PROJECT'/topics/your-topic-name",
+                     "outputFormat": "PLAINTEXT",
+                     "secretStoreUrl": "http(s)://host:port/path/to/credentials",
+                     "vaultToken": "your-token"
+                 }
+             }
+         }
+        '
+        "${TEMPLATES_LAUNCH_API}"
+    ```
+
+## AVRO format transferring.
+This template contains an example Class to deserialize AVRO from Kafka and serialize it to AVRO in Pub/Sub.
```

Comment:
I'm not sure I understand the purpose of this, won't this just end up re-serializing to the same byte array? And in that case couldn't we just forward the value byte array directly instead?

Maybe I'm missing something, could you clarify?

Reply:
Yes, as you mentioned we can read and write bytes without any serialization and deserialization. In our case, we want to give an example of Avro usage for users who will work with this template. It helps easily modify the class to deserialize data and, if it is needed, implement any transformation using this class. 

I updated our example to be more abstract. Now it is more obvious that users need to make changes to our class for working with each specific dataset.

Reply:
But even with the updated language, the user would just deserialize the byte array to an instance `MyAvroClass` and then reserialize it back to the same byte array to send to PubSub. If we want to show an example of deserializing and serializing, it should at least do some processing, or convert the message to another format. Otherwise I feel this could just confuse users.

Reply:
I understand where the frustration for someone looking for an example might come from. I included clarity in the readme where a user would plug in their data transforms, calling out those steps with [OPTIONAL TO IMPLEMENT] and providing a link to Beam transforms that go into details of transforms.